src/geometry.rs

```diff
@@ -356,6 +357,42 @@ impl Geometry {
     }
 }
 
+impl fmt::Display for Geometry {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let vertices: Vec<_> = self
+            .vertices
+            .iter()
+            .enumerate()
+            .map(|(i, v)| format!("{}: {}", i, v))
+            .collect();
+        let faces: Vec<_> = self
+            .faces
+            .iter()
+            .enumerate()
+            .map(|(i, f)| format!("{}: {}", i, f))
+            .collect();
+        let normals: Vec<_> = self
+            .normals
+            .iter()
+            .enumerate()
+            .map(|(i, n)| format!("{}: ({}, {}, {})", i, n.x, n.y, n.z))
+            .collect();
+        write!(
+            f,
+            "Vertex count: {}, Normal count: {}, Face count: {}\n\
+             Vertices: {:?}\n\
+             Normals: {:?}\n\
+             Faces: {:?}",
```

Comment:
I agree it might be irrelevant to the rest of the PR but it was a great tool for debugging. The `Debug` dump wasn't easy enough to read, whereas this was and helped me to find errors in both, the functions and their tests. I believe this will be a very useful way of tracking possible errors in the upcoming geometry modifiers.

I don't mind splitting the current PR into two, just have no idea how.